libraries/cms/application/site.php

```diff
@@ -651,6 +621,25 @@ protected function initialiseApp($options = array())
 	}
 
 	/**
+	 * Language check
+	 *
+	 * @param  string  Language to check.
+	 *
+	 * @return  mixed  string  The passed language if it is an available language.
+	 *                 null    If the language is not an available language.
+	 *
+	 * @since   3.4.2
+	 */
+	protected function checkLang($lang)
```

Comment:
private methods are fine and don't fall under B/C. protected (and obviously public) methods indeed have to be backward compatible if the class is meant to be used/extended. Which means basically everything in libraries folder.